source/fundamentals/data-formats/pojo-customization.txt

```diff
@@ -10,78 +10,653 @@ POJO Customization
+
+Customize a PojoCodecProvider
+-----------------------------
+
+This section shows you how to specify your data conversion logic and POJO
+classes with a ``PojoCodecProvider``. The ``PojoCodecProvider`` is an
+implementation of the ``CodecProvider`` interface that specifies the
+Codecs to use in data conversion. Use this implementation when performing
+data conversion between BSON and POJOs.
+
+You can create a ``PojoCodecProvider`` instance using the
+``PojoCodecProvider.builder()`` method. You can also chain methods to the
+builder to register any of the following:
+
+- individual POJO classes
+- package names that contain POJO classes
+- instances of ``ClassModel`` that describe conversion logic for a specific
+  POJO class
+
+The following example shows how you can specify the POJOs in a package named
+"org.example.pojos" and add the ``PojoCodecProvider`` to a ``CodecRegistry``:
 
+.. code-block:: java
+   :copyable: true
+   :emphasize-lines: 7-8
+
+   import org.bson.codecs.configuration.CodecProvider;
+   import org.bson.codecs.configuration.CodecRegistry;
+   import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+   import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+   import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+
+   CodecProvider pojoCodecProvider = PojoCodecProvider.builder().register("org.example.pojos").build();
+   CodecRegistry pojoCodecRegistry = fromRegistries(getDefaultCodecRegistry(), fromProviders(pojoCodecProvider));
+
+   // Call withCodecRegistry(pojoCodecRegistry) on an instance of MongoClient, MongoDatabase, or MongoCollection
+
+See the :java-docs:`PojoCodecProvider.Builder API documentation </apidocs/bson/org/bson/codecs/pojo/PojoCodecProvider.Builder.html>`
+for more information.
+
+.. _classmodel:
 
 ClassModel
-----------
+~~~~~~~~~~
+
+A ``ClassModel`` instance stores data conversion information about a specific
+POJO class. It contains a list of ``PropertyModel`` instances which describe
+the property fields of the POJO, whether to convert fields, and optionally,
+``Codecs`` to convert them with.
+
+A ``ClassModel`` contains the following fields:
+
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+
+   * - Field Name
+     - Description
+
+   * - Name
+     - The POJO class name to associate with the ``ClassModel``.
 
-- Description
-- Table of properties
-- Link to Conventions section for example of ClassModelBuilder
+   * - InstanceCreatorFactory
+     - Contains a new instance factory that creates new instances of the POJO.
+       By default, it requires the POJO to have an empty constructor.
+
+   * - PropertyModels
+     - Contains a list of ``PropertyModel`` instances that specify how to
+       convert data to and from BSON for a field in the POJO.
+
+   * - IdPropertyModelHolder
+     - Specifies the POJO field that corresponds to the document id field. Optional.
```

Comment:
```suggestion
     - Specifies the POJO field that corresponds to the document ``_id`` field. Optional.
```
I assume this is talking about the MongoDB default _id field. Would this be more correct, or does the Java driver use a different convention?